### PR TITLE
add rbac checks in all the pages

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/HAC-3430

## Description

Add access checks to the pages and show an no access state if user does not have enough access to view or perform an primary action on a page.

## Type of change

- [x] Feature

## Screen shots / Gifs for design review 

Add Integration test page:
Screenshot 2023-04-01 at 12 51 58 AM

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc: @beaumorley @Misjohns @christianvogt

### CODE REVIEW

File: src/pages/ApplicationsPage.tsx [hunk 1]
```diff
@@ -2,14 +2,21 @@ import React from 'react';
...
 import { HeadTitle } from '../components/HeadTitle';
 import NamespacedPage from '../components/NamespacedPage/NamespacedPage';
+import PageAccessCheck from '../components/PageAccess/PageAccessCheck';
 import { useQuickstartCloseOnUnmount } from '../hooks/useQuickstartCloseOnUnmount';
+import { ApplicationModel } from '../models';
+import { AccessReviewResources } from '../types';
 
 const ApplicationsPage = () => {
   useQuickstartCloseOnUnmount();
+  const accessReviewResources: AccessReviewResources = [{ model: ApplicationModel, verb: 'get' }];
```
Comment (rottencandy) [inline src/pages/ApplicationsPage.tsx:12]: Not entirely sure about this, but for list pages, should we be using the `list` verb instead? Although we are actually doing a `watch` :thinking:

File: src/pages/ImportPage.tsx [hunk 2]
```diff
@@ -13,14 +16,20 @@ const ImportPage: React.FunctionComponent = () => {
   const applicationBreadcrumbs = useApplicationBreadcrumbs(applicationName);
 
   const title = applicationName ? 'Add component' : 'Create application';
+  const accessReviewResources: AccessReviewResources = [
+    { model: ApplicationModel, verb: 'create' },
+    { model: ComponentModel, verb: 'create' },
```
Comment (rottencandy) [inline src/pages/ImportPage.tsx:21]: Might need to conditionally add application check based on `applicationName`

File: src/pages/ApplicationsPage.tsx (hunk 1 above)
Comment (karthikjeeyar) [reply src/pages/ApplicationsPage.tsx:12]: Ah I see. For list page maybe we can add both `list`, `watch`

File: src/components/PageAccess/NoAccessState.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <EmptyState
+      className={css('app-empty-state', className)}
+      variant={EmptyStateVariant.full}
+      {...props}
+      data-test="no-access-state"
+    >
+      <EmptyStateIcon className={css('app-empty-state__icon ')} icon={LockIcon} />
+      <Title className="pf-u-mt-lg" headingLevel="h2" size="lg">
+        {title || 'Access permission Denied'}
```
Comment (abigaeljamie) [inline src/components/PageAccess/NoAccessState.tsx:43]:
```suggestion
        {title || 'Access permissions needed'}
```

File: src/components/PageAccess/NoAccessState.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      </Title>
+      <EmptyStateBody>
+        {body ||
+          `You don't have access to this page, ask your administrator to adjust your permissions.`}
```
Comment (abigaeljamie) [inline src/components/PageAccess/NoAccessState.tsx:47]:
```suggestion
          `To access this page, ask your administrator to adjust your permissions.`}
```

**Review by abigaeljamie — CHANGES_REQUESTED** [review]
I left some updated microcopy for the "No access" empty state to align with the microcopy pattern we have on PatternFly: https://www.patternfly.org/v4/components/empty-state/design-guidelines 

The goal of that pattern is to sound a little friendlier and shift the focus to what the user needs (vs. what they don't have).

**Comment by karthikjeeyar** [comment]
Thanks @abigaeljamie. I have updated the content and screenshots.

**Review by abigaeljamie — APPROVED** [review]
Thank you, @karthikjeeyar ! This looks great 🙂

**Pull request merged**